posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,799 @@
+
+**Details:** We had three specific aspirations identified in the compiler team blog post:
+
+1. Better integration between Cargo and debuggers, specifically in regards to starting and configuring the debugger. 
+   - We have not made progress on this aspiration.
+2. Improve expression evaluation support in various debuggers.
+   - wg-debugging has had discussions during our status & design meetings on how this could be achieved, but we have not made concrete progress on this aspiration.
+3. Revise our debugger extension architecture for rendering Rust data structures.
+   - [RFC 3191](https://github.com/rust-lang/rfcs/pull/3191) has been approved and implemented on nightly toolchains but not yet stabilized.
+
+Regarding new contributors:
+RFC 3191 was written and implemented by a new contributor (@ridwanabdillahi).
+
+Regarding prioritization and focus:
+There isn't much focus happening here currently.
+
+### Faster Builds Aspirations (👩‍💻, 🛠️)
+
+### Incremental Compilation Aspirations
+
+<!-- https://hackmd.io/@rust-compiler-team/B1D7CiX_c -->
+
+Progress:
+The Rust project has made progress on this ambition since the start of the year
+
+Goals:
+we had no goals for this ambition planned for this year, but we made ad-hoc progress on the problem itself.
+we think a solution will be available in the next two years, but not sooner than six months.
+
+How it started:
+at start of 2022, we knew a problem existed, but we did not yet have a specific goal in mind for solving the problem.
+
+How it's going:
+we would like help deciding what to do next
+we have Rust contributors who have agreed to help with the units of work that we have identified
+
+Details:
+
+
+
+We have refactored the global HIR visitor to reduce the amount of accesses to HIR. Using this, we managed to reduce incremental reuse and avoid some recomputations.
+
+We have laid down the first steps for implementing “salsa jars” in the rustc query system. Those first steps are being implemented by a contributor, but the future plans are not fully designed yet.
+
+We have established the constraints for extending the query system with “salsa entities”. The design is much more subtle than anticipated, and will take longer than planned.
+
+Integration of lowering into the query system is still in review. This blocks progress on making name resolution incremental.
+
+
+Regarding new contributors:
+The progress has been made in part thanks to Miguel Guarniz (kckeiks).
+
+
+Regarding prioritization and focus:
+One of the owners, cjgillot started a large refactor of lifetime resolution.
+This refactor allowed for faster progress in fixing a few old bugs.
+
+
+
+### Inter-crate Sharing Aspirations
+
+<!-- https://hackmd.io/zad1BjIoR8S4IfE49z-xYg -->
+<!-- (no survey response as of 2022-07-19) -->
+
+No survey response was provided; we assume no progress has been made. This aspiration may have been abandoned.
```

Comment:
I think we should follow up and confirm this so we can make a definitive statement.

Reply:
Okay I'll reach out. (I think I might have already, but I'll double-check in any case.)

Reply:
Fixed in 18fd548b7da90bfb8cf9c6c84467c96d97c1d370